sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/integrationtest/BeamSqlBuiltinFunctionsIntegrationTestBase.java

```diff
@@ -152,12 +215,14 @@ private static ExpressionTestCase of(
     private transient List<ExpressionTestCase> exps = new ArrayList<>();
 
     public ExpressionChecker addExpr(String expression, Object expectedValue) {
-      // Because of erasure, we can only automatically infer non-parameterized types
       TypeName resultTypeName = JAVA_CLASS_TO_TYPENAME.get(expectedValue.getClass());
       checkArgument(
           resultTypeName != null,
-          "Could not infer a Beam type for %s."
-              + " Parameterized types must be provided explicitly.");
+          String.format(
+              "The type of the expected object '%s' is unknown in 'addExpr(String %s, Object %s)'"
```

Comment:
this `addExpr` example can be confusing now, e.g. if you call this method with something like `addExpr("++b ", Arrays.<MyObj> toList(myObj1, ...)`, then the error message would be something like `addExpr(String ++b, Object [com.blah.myObj@af213])` which looks bad. I'd rather just leave the parameter names.

Reply:
Updated.

I kept the first %s for now.